addon/record-data.js

```diff
@@ -239,7 +241,9 @@ export default class M3RecordData {
     }
 
     if (CUSTOM_MODEL_CLASS) {
-      this.globalM3CacheRD[this.id] = this;
+      if (!this._baseRecordData && !this._parentRecordData && this.id) {
```

Comment:
Let's avoid boolean coercion if possible.

```suggestion
      if (this._baseRecordData !== undefined && this._parentRecordData !== undefined && this.id !== undefined) {
```

Also, I don't grok why `this.id` could be falsey.

Reply:
commented above, lets combine the discussion there